libs/hwdrivers/src/rplidar/src/rplidar_driver.cpp

```diff
@@ -349,10 +570,7 @@ u_result RPlidarDriverSerialImpl::ascendScanData(rplidar_response_measurement_no
     }
 
     // all the data is invalid
-    if (i == count){
-        delete[] tmpbuffer;
-        return RESULT_OPERATION_FAIL;
-    }
+    if (i == count) return RESULT_OPERATION_FAIL;
 
     //Tune tail
     for (i = count - 1; i >= 0; i--) {
```

Comment:
yeah, I agree with you about the warnings. Not notice it without clang and introduce it into sdk at first.  I will try to modify it at next version sdk.